File: ERPServer/Shared/Shared/Entities/Accouting/TreasuryDTO.cs
﻿using Shared.Entities.Shared;
using Shared.Enums;

namespace Shared.Entities.Accouting
{
    public class TreasuryDTO : BaseDTO
    {
        public string Number { get; set; }
        public string Date { get; set; }
        public AccountTypeEnum AccountTypeId { get; set; }
        public long? ClientVendorId { get; set; }
        public string BeneficiaryName { get; set; }
        public PaymentMethodEnum? PaymentMethodId { get; set; }
        public decimal InComing { get; set; }
        public decimal OutComing { get; set; }
        public string RefNo { get; set; }
        public string Notes { get; set; }
        public bool IsCancel { get; set; }
        public bool IsBilled { get; set; }

    }
}

File: erp-app/src/app/modules/accounting/models/treasury.dto.ts
import { AccountTypeEnum } from "src/app/shared/enums/account-type.enum";
import { PaymentMethodEnum } from "src/app/shared/enums/payment-method.enum";
import { TransactionTypeEnum } from "src/app/shared/enums/transaction-type.enum";
import { BaseDTO } from "src/app/shared/models/base-dto.model";

export class TreasuryDTO extends BaseDTO {

    number: number;
    date: string ;
    accountTypeId: AccountTypeEnum;
    clientVendorId: number | null;
    beneficiaryName: string | undefined;
    //transactionTypeId: TransactionTypeEnum;
    paymentMethodId: PaymentMethodEnum;
    outComing: number = 0;
    inComing: number = 0;
    refNo: string;
    notes: string;
    isBilled: boolean;
}